gradio/utils.py

```diff
@@ -384,6 +384,14 @@ def run_coro_in_background(func: Callable, *args, **kwargs):
     return event_loop.create_task(func(*args, **kwargs))
 
 
+def async_iteration(iterator):
+    try:
+        return next(iterator)
+    except StopIteration:
+        # raise a ValueError here because co-routines can't raise StopIteration themselves
```

Comment:
Would `StopAsyncIteration` work here? https://stackoverflow.com/questions/50165734/native-coroutine-and-stopiteration